lib/workers/branch/lock-files.js

```diff
@@ -123,6 +123,34 @@ async function writeExistingFiles(config) {
         upath.join(basedir, 'package.json'),
         JSON.stringify(massagedFile)
       );
+
+      // if you have some reference to local files, copy them
+      const toCopy = listLocalLibs(massagedFile.dependencies);
+      toCopy.push(...listLocalLibs(massagedFile.devDependencies));
+      if (toCopy.length !== 0) {
+        logger.debug(`listOfNeededLocalFiles files to copy: ${toCopy}`);
+        await Promise.all(
+          toCopy.map(async localPath => {
+            const resolvedLocalPath = path.resolve(basedir, localPath);
+            // at the root of local Lib we should find a package.json and yarn will use it to update yarn.lock file
+            const resolvedRepoPath = upath.join(
+              resolvedLocalPath.substring(config.tmpDir.path.length + 1),
+              'package.json'
+            );
+            const fileContent = await platform.getFile(resolvedRepoPath);
+            if (fileContent !== null) {
+              await fs.outputFile(
+                upath.join(resolvedLocalPath, 'package.json'),
```

Comment:
Why is 'package.json' joined here?

Reply:
First I wanted to copy the folder, but I found out that to generate the yarn.lock we just need the package.json of the local lib. => So we get only 1 file! 👍

Reply:
That's really good. What about the case where you have the binary files? Are those normally accompanied with a `package.json` beside them?

Reply:
Inside the binary file you have the package.json (not beside).
But I was not able to download and write to disk the binary file! (I sent some hours there!)

I would say that this is a frist step to local package, then if someone request tgz we will enhance...